hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/scope/HandleHederaNativeOperations.java

```diff
@@ -184,4 +185,9 @@ public boolean checkForCustomFees(@NonNull final CryptoTransferTransactionBody o
     public SortedSet<Key> authorizingSimpleKeys() {
         return context.keyVerifier().authorizingSimpleKeys();
     }
+
+    @Override
+    public TransactionID getTransactionID() {
+        return context.body().transactionIDOrElse(TransactionID.DEFAULT);
```

Comment:
Based on the documented contract for this method,
```
Returns the {@link TransactionID} for the top level transaction.
```
...shouldn't this be `context.body.transactionIDOrThrow()`?

Because a top-level transaction is either (1) a HAPI transaction or (2) a triggered `ContractCall` that was scheduled with `wait_for_expiry=true`; and in both cases, it must have a `TransactionID` set.

Even given a `ContractCall` that uses HSS to sign a `wait_for_expiry=false` scheduled `ContractCall`, the scheduled dispatch for the triggered `ContractCall` will have a `TransactionID`.

So if `OrElse()` is necessary to avoid a NPE, it may be hiding a bug. 🤔